File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/units/Centimeter.java
package org.firstinspires.ftc.teamcode.units;

public class Centimeter implements Distance {
    private final double value;

    public Centimeter(double value) {
        this.value = value;
    }

    @Override
    public double value() {
        return value;
    }

    @Override
    public double toCM() {
        return value;
    }

    @Override
    public double toInches() {
        return value/2.54;
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/units/Centimeter.kt
package org.firstinspires.ftc.teamcode.units

import org.firstinspires.ftc.robotcore.external.navigation.DistanceUnit

open class Centimeter(override val value: Double) : Distance() {
    override val unit = Static.unit
    companion object Static {
        val unit = DistanceUnit.CM
    }

    constructor(dist: Distance): this(dist.toUnit(Static.unit).value)

    override fun toCM(): Centimeter {
        return this
    }

    override fun toInches(): Inches {
        return Inches(this)
    }
}
